### PR TITLE
[BS-295][LB] Review use of HeaderCarrier to pass headers to connector

### PR DESCRIPTION
**Pull request by udinobi**

### CODE REVIEW

File: app/uk/gov/hmrc/breathingspaceifproxy/connector/ConnectorHelper.scala [hunk 1]
```diff
@@ -16,89 +16,52 @@
...
 
-// TODO: should not be composing responses to PEGA here in the IF connector!
 trait ConnectorHelper extends HttpErrorFunctions with Logging {
 
-  val ifCorrelationIdHeaderName = "CorrelationId"
-  val ifRequestTypeHeaderName = "OriginatorId"
-  val ifStaffIdHeaderName = "UserId"
-
-  def composeResponseFromIF(response: HttpResponse)(implicit url: Url, headerSet: RequiredHeaderSet): Future[Result] = {
+  def composeResponse(response: HttpResponse)(implicit requestId: UUID, url: Url): Result = {
```
Comment (udinobi) [inline app/uk/gov/hmrc/breathingspaceifproxy/connector/ConnectorHelper.scala:34]: This method will be completely removed at the next ticket where the connector refactoring will take place.

**Review by LokeshSrinivasa — APPROVED** [review]
accepted

File: conf/application.conf [hunk 2]
```diff
@@ -138,7 +137,6 @@ mapping {
   }
 }
 
-# should these not be whitelisting our own request headers that PEGA will send to us (not IF headers)?
 httpHeadersWhitelist += ${mapping.if.correlation-id}
 httpHeadersWhitelist += ${mapping.if.request-type}
 httpHeadersWhitelist += ${mapping.if.staff-id}
```
Comment (glidester) [inline conf/application.conf:142]: Still think this is wrong. Surly we need to white list our header keys that PEGA will send to us (so they are allowed when the controller calls the HeaderCarrierConverter function)?

**Review by glidester — CHANGES_REQUESTED** [review]

File: conf/application.conf (hunk 2 above)
Comment (udinobi) [reply conf/application.conf:142]: But we'll be using these keys not to white list the headers PEGA is sending to us, but to white list the headers we send to IF while generating the HeaderCarrier. Hope you get it now.

Comment (glidester) [reply conf/application.conf:142]: Seems a bit of a backwards way to do it IMO but ok

**Review by glidester — APPROVED** [review]

**Pull request merged**